### PR TITLE
Encryption Scope SAS

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Adding encryption sas to blob, account, and identity sas

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/sas/account.go [hunk 1]
```diff
@@ -25,13 +25,15 @@ type UserDelegationCredential = exported.UserDelegationCredential
...
-	IPRange       IPRange   `param:"sip"`
-	ResourceTypes string    `param:"srt"` // Create by initializing AccountResourceTypes and then call String()
+	Version         string    `param:"sv"`  // If not specified, this format to SASVersion
+	Protocol        Protocol  `param:"spr"` // See the SASProtocol* constants
+	StartTime       time.Time `param:"st"`  // Not specified if IsZero
+	ExpiryTime      time.Time `param:"se"`  // Not specified if IsZero
+	Permissions     string    `param:"sp"`  // Create by initializing AccountPermissions and then call String()
+	IPRange         IPRange   `param:"sip"`
+	ResourceTypes   string    `param:"srt"` // Create by initializing AccountResourceTypes and then call String()
+	Identifier      string    `param:"si"`
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/sas/account.go:35]: I don't think Signed Identifiers(si) are used in Account SAS creation. Also, this field is not used anywhere in the code below. So, it should be removed.

File: sdk/storage/azblob/sas/query_params.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ const (
 
 var (
 	// Version is the default version encoded in the SAS token.
-	Version = "2020-02-10"
+	Version = "2020-12-06"
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/sas/query_params.go:26]: The version should be the STG 85/86 version in my opinion.
```suggestion
	Version = "2021-12-02"
```

File: sdk/storage/azblob/sas/service.go [hunk 3]
```diff
@@ -103,7 +104,8 @@ func (v BlobSignatureValues) SignWithSharedKey(sharedKeyCredential *SharedKeyCre
 		string(v.Protocol),
 		v.Version,
 		resource,
-		snapshotTime,         // signed timestamp
+		snapshotTime, // signed timestamp
+		v.EncryptionScope,
 		v.CacheControl,       // rscc
 		v.ContentDisposition, // rscd
 		v.ContentEncoding,    // rsce
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/sas/service.go:111]: As per the [documentation](https://learn.microsoft.com/en-us/rest/api/storageservices/create-service-sas#version-2020-12-06-and-later), `v.ContentType // rsct` is removed from version 2020-12-06 and later for creating signature string. You can check if this is actually the case or is it a documentation bug.

File: sdk/storage/azblob/CHANGELOG.md [hunk 4]
```diff
@@ -6,7 +6,8 @@
 * Added support for [Cold tier](https://learn.microsoft.com/azure/storage/blobs/access-tiers-overview?tabs=azure-portal).
 * Added `CopySourceTag` option for `UploadBlobFromURLOptions`
 * Added [FilterBlobs](https://learn.microsoft.com/rest/api/storageservices/find-blobs-by-tags-container) by tags API for container client.
-* Added `System` option to `ListContainersInclude` to allow listing of system containers. 
+* Added `System` option to `ListContainersInclude` to allow listing of system containers.
+* Updated the SAS Version to `2020-12-06` and added `Encryption Scope` to Account SAS, Blob SAS, and User Delegation SAS
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/CHANGELOG.md:10]:
```suggestion
* Updated the SAS Version to `2021-12-02` and added `Encryption Scope` to Account SAS, Service SAS, and User Delegation SAS 
```

Comment (seanmcc-msft) [reply sdk/storage/azblob/CHANGELOG.md:10]: We should make sure we are sending the 2021-12-02 SAS version to the service, it should match the x-ms-version.

File: sdk/storage/azblob/sas/query_params.go (hunk 2 above)
Comment (siminsavani-msft) [reply sdk/storage/azblob/sas/query_params.go:26]: That's right! Thanks for the call out!

File: sdk/storage/azblob/sas/account.go (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/azblob/sas/account.go:35]: I agree with @souravgupta-msft, I am not seeing Signed Identifiers in the REST docs for account SAS - https://learn.microsoft.com/en-us/rest/api/storageservices/create-account-sas

File: sdk/storage/azblob/sas/service.go (hunk 3 above)
Comment (siminsavani-msft) [reply sdk/storage/azblob/sas/service.go:111]: I believe that is a bug, I tried to remove it and get 'Authentication Failed'. I double checked the string to sign in the .NET SDK and it matches with what we have here.

https://github.com/Azure/azure-sdk-for-net/blob/e8f203346cbc8135aac9ce250853fd4cbed9d75d/sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs#L350

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**